### PR TITLE
Mkirk/session reset

### LINKED ISSUE #1939: "No available session for contact" alerts
- [X] I have searched open and closed issues for duplicates

----------------------------------------

### Bug description
No available session with contact alerts in the conversation thread.  

### Steps to reproduce
- No reliable steps to repro
- Send message

**Actual result:** See alert of "No availble session with contact."
**Expected result:** No alert

### Screenshots
screen shot 2017-03-22 at 1 27 15 pm

### Device info

**Device**: iPhone 5
**iOS version**: 10.2.1
**Signal version:** 2.8.3

### Link to debug log

https://gist.github.com/281b487b35b9fa49df836b727680ce3e

### PR DESCRIPTION
**Pull request by michaelkirk**
Otherwise the other devices for that contact will send you undecryptable
messages.

fixes https://github.com/WhisperSystems/Signal-iOS/issues/1939#issuecomment-294005518

PTAL @charlesmchen

### CODE REVIEW

File: src/Messages/TSMessagesManager.m [hunk 1]
```diff
@@ -582,7 +582,8 @@ - (void)handleEndSessionMessageWithEnvelope:(OWSSignalServiceProtosEnvelope *)en
     }];
 
     dispatch_async([OWSDispatch sessionStoreQueue], ^{
-        [[TSStorageManager sharedManager] deleteAllSessionsForContact:endSessionEnvelope.source];
+        [[TSStorageManager sharedManager] deleteSessionForContact:endSessionEnvelope.source
+                                                         deviceId:endSessionEnvelope.sourceDevice];
```
Comment (charlesmchen) [inline src/Messages/TSMessagesManager.m:586]: So we were being overzealous in our deletion?  Makes sense.

**Review by charlesmchen — APPROVED** [review]
LGTM

File: src/Messages/TSMessagesManager.m (hunk 1 above)
Comment (michaelkirk) [reply src/Messages/TSMessagesManager.m:586]: Right, fixes the scenario wherein:

* Alice.1 has a session with Bob.1 and Bob.2
* Bob.1 sends an EndSession to Alice
* Thus Alice.1 deletes her sessions with Bob.1 and Bob.2
* Bob.2 sends Alice1 "hi" (with the existing session - as a regular WM, not a PKWM)
* Alice has no session with Bob.2 so cannot decrypt the message.

**Comment by michaelkirk** [comment]
Closing for an approach more in line with Signal-Desktop/Android behavior.

**Pull request closed without merging**